util/src/main/java/com/farao_community/farao/util/SensitivitySecurityAnalysisService.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    public static SensitivitySecurityAnalysisResult runSensitivity(Network network,
+                                                                   Crac crac,
+                                                                   ComputationManager computationManager) {
+        //prepare range actions: pst ( and hvdc in the future )
+        List<TwoWindingsTransformer> twoWindingsTransformers = getPstInRangeActions(network, crac.getRangeActions());
+
+        //contingency = null : pre-contingency sensitivity computation analysis
+        LOGGER.info("Running pre contingency sensitivity computation");
+        SensitivityComputationResults precontingencyResult = runSensitivityComputation(network, crac, twoWindingsTransformers);
+
+        //get result per contingency
+        LOGGER.info("Calculating sensitivity result per contingency");
+        Map<Contingency, SensitivityComputationResults> contingencySensitivityComputationResultsMap = new HashMap<>();
+
+        String initialVariantId = network.getVariantManager().getWorkingVariantId();
+        try (FaraoVariantsPool variantsPool = new FaraoVariantsPool(network, initialVariantId)) {
+            variantsPool.submit(() -> crac.getContingencies().forEach(contingency -> {
+                try {
+                    LOGGER.info("Running post contingency sensitivity computation for contingency '{}'", contingency.getId());
+                    String workingVariant = variantsPool.getAvailableVariant();
+                    network.getVariantManager().setWorkingVariant(workingVariant);
+                    applyCracContingency(network, computationManager, contingency);
+
+                    SensitivityComputationResults sensiResults = runSensitivityComputation(network, crac, twoWindingsTransformers);
+                    contingencySensitivityComputationResultsMap.put(contingency, sensiResults);
+
+                    variantsPool.releaseUsedVariant(workingVariant);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            })).get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (ExecutionException e) {
+            throw new FaraoException(e);
+        }
+        network.getVariantManager().setWorkingVariant(initialVariantId);
+
+        //return SensitivitySecurityAnalysisResult
+        return new SensitivitySecurityAnalysisResult(precontingencyResult, contingencySensitivityComputationResultsMap);
+    }
+
+    private static void applyCracContingency(Network network, ComputationManager computationManager, Contingency contingency) {
+        contingency.getNetworkElements().forEach(contingencyElement -> applyCracContingencyElement(network, computationManager, contingencyElement));
+    }
+
+    private static void applyCracContingencyElement(Network network, ComputationManager computationManager, NetworkElement contingencyElement) {
```

Comment:
Same remark, I would rename this function.

Reply:
.